src/main/java/com/phraktle/histodiff/HistoDiff.java

```diff
@@ -20,18 +24,37 @@
     public static void main(String[] args) throws IOException {
         if (args.length < 2) {
             System.out
-                    .println("Usage: HistoDiff file1 file2 [sortBy] [threshold]");
+                    .println("Usage: HistoDiff file1/url1 file2/url2 [sortBy] [threshold]");
             System.exit(1);
         }
-        File file1 = new File(args[0]);
-        File file2 = new File(args[1]);
+        File file1 = parseFileArg(args, 0);
+        File file2 = parseFileArg(args, 1);
         int sortBy = parseArg(args, 2);
         int threshold = parseArg(args, 3);
 
         dump(sort(sortBy,
                 filter(sortBy, threshold, diff(parse(file1), parse(file2)))));
     }
 
+    static File parseFileArg(String[] args, int idx) {
+        if (args[idx].startsWith("http")) {
+            try {
+                File targetFile = File.createTempFile("histodiff-temp-file-", ".tmp");
+                targetFile.deleteOnExit();
+                URL httpTarget = new URL(args[idx]);
+                try (InputStream in = httpTarget.openStream()) {
+                    Files.copy(in, targetFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+                }
+                return targetFile;
+            } catch (Exception e) {
```

Comment:
Exceptions should not be caught here, declare as `throws` (as well as in `main`). Don't return `null`.